git-cheatsheet.md

```diff
@@ -0,0 +1,40 @@
+# git cheatsheet
+This page gathers some instructions around the git functionalities we'll use throughout the workshop, so feel free to consult at any time!
+
+## Clone
+
+For full documentation on this command check the [git docs](https://www.git-scm.com/docs/git-clone).  
+
+* To clone a repository follow the instructions on this [GitHub article](https://help.github.com/en/articles/cloning-a-repository)
+
+## Create a branch
+
+This [Git Branching doc](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) is a very comprehensive guide to Basic branching and merging, and it's worth the read if you're new to git and it's architecture.
+Also, worth to know about [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/), which is one of the most used branching models (either pure GitFlow or some variation of it).  
+
+To create a new branch:  
+
+1. `git checkout -b <your-branch-name>`
+
+## Commit
+
+For more info on commits, see [the docs](https://git-scm.com/docs/git-commit) and this useful article on [Saving changes](https://www.atlassian.com/git/tutorials/saving-changes) on git.  
+
+To commit new changes:
+1. Prepare the new files for the comming by adding them to the staging area:  
+  1.1 `git add -all`
+2. Then, commit your code via `git commit -a -m <commit-message>`  
```

Comment:
`git commit -a` and using `git add` to stage specific changes are mutually exclusive workflows, and I also think we should steer clear of `-m` and instead get people to write commit messages in an editor and treat them as a kind of documentation.

Perhaps we could tell people about `git add <filepath>` as well, and then recommend they use `git commit` on its own to commit staged changes.

Reply:
Oops, you're right, I'll remove the -a and add the `git add <file>` (then list the -all as an alternative for this first round)

On the commit message, I wanted to get them to stay away from terminal editors, that's why I want with -m, but I'll add some info there and assume they'll open a vim and behave on it :)

Thanks! :)

Reply:
Hmm, that's a good point about console editors. Perhaps we could (not necessarily in this PR) include some guidance on setting up VS Code as git's editor of choice?

Reply:
Sorry, somehow I missed your response. We can add instructions in the git cheatsheet and reference in the setting up section, what do you think?